### ISSUE TITLE
[Bug]: S120 Solar Wall Light Cam: Error no currentSnapshot available

### ISSUE THREAD

**Issue by mgeorgia** [homebridge-eufy-security/plugin#639]
### What happened?

I have two Eufy devices - a doorbell cam and a S120 solar wall light cam.  The doorbell cam (including snapshot behavior) is working as expected.

However, the snapshot behavior for the S120 solar wall light cam is not working as expected.  When I open the preview page in HomeKit (where it shows the snapshots for all my cameras), for the doorbell cam I see the last snapshot taken by the camera.  For the S120 solar wall light cam, I see "Snapshot Unavailable Unable to fetch snapshot".  If I click the S120 camera to view the live stream, the live stream works properly.

The logs sent below are with debug off.  I can send debug logs if needed.

I'm running the latest stable homebridge and plugin release on Raspbian (bookworm). NodeJs v20.11.0.  The drop-downs in the bug report may be out of date?

I changed my codec settings for the wall light cam to try to improve performance per a reddit thread I found (https://www.reddit.com/r/homebridge/comments/18cd0vd/need_help_with_adding_eufy_s100_wall_light_and/), however, the issue persists with both the default as well as these updated codec settings.

Homebridge: v1.8.4
Plugin: v4.3.0
NodeJs: v20.11.0

### Device Type

Floodlight (Supported)

### Plugin Version

other (UnSupported)

### HomeBridge Version

other (UnSupported)

### NodeJS Version

v20 (Supported)

### Operating System

Raspbian 10+ (Supported)

### Relevant log output

```shell
[10/16/2024, 08:44:43]  [EufySecurity]  ERROR   Solar Wall Light Cam

 Error  No currentSnapshot available
error stack:
  • SnapshotManager.ts  SnapshotManager.getSnapshotFromCache
        SnapshotManager.ts:299
  • SnapshotManager.ts  SnapshotManager.getSnapshotBuffer
        SnapshotManager.ts:241
  • SnapshotManager.ts  SnapshotManager.getSnapshotBufferResized
        SnapshotManager.ts:192
  • streamingDelegate.ts        StreamingDelegate.handleSnapshotRequest
        streamingDelegate.ts:107
  • CameraController.ts
        CameraController.ts:1032
  • 

  • CameraController.ts CameraController.handleSnapshotRequest
        CameraController.ts:1011
  • Accessory.ts        Bridge.Accessory.handleResource
        Accessory.ts:1949
  • events      HAPServer.emit
        events:518
  • domain      HAPServer.emit
        domain:488
```

### config

```shell
{
            "platform": "EufySecurity",
            "name": "EufySecurity",
            "username": "",
            "password": "",
            "country": "US",
            "deviceName": "",
            "cameras": [
                {
                    "serialNumber": "",
                    "rtsp": false,
                    "enableButton": true,
                    "motionButton": true,
                    "lightButton": false,
                    "snapshotHandlingMethod": 3,
                    "delayCameraSnapshot": true,
                    "videoConfig": {
                        "vcodec": "copy",
                        "useSeparateProcesses": true
                    }
                },
                {
                    "serialNumber": "",
                    "rtsp": false,
                    "snapshotHandlingMethod": 3,
                    "immediateRingNotificationWithoutSnapshot": true,
                    "delayCameraSnapshot": true,
                    "refreshSnapshotIntervalMinutes": 0
                }
            ],
            "_bridge": {
                "username": "",
                "port": 51352
            },
            "ignoreStations": [
                "",
                ""
            ]
        }
```

**Comment by mgeorgia** [comment]
> Did you check this [Common Issues](https://github.com/homebridge-eufy-security/plugin/wiki/Common-Issues) pages ?

yes, I did.  I did not see any relevant common issues pertaining to this particular problem I'm having.

**Comment by mgeorgia** [comment]
Additional note.  I configured notifications for motion for this camera as well, and the motion notification provided by HomeKit also has a thumbnail image showing the same "Snapshot Unavailable" message.

This happens even after changing the camera notification settings to include the thumbnail by setting Notification->Notification Type="With Thumbnail".

**Comment by ArtB23** [comment]
Hi, I’m experiencing the same issue with two S120 and Homebridge. Everything is updated to the latest firmware, and I’ve checked various settings. I managed to get a single snapshot once, but I’ve never been able to get it again. I’ve never managed to get the livestream.

**Comment by mar894q** [comment]
I have the same issue with my cameras. Did you find a solution?

**Comment by lenoxys** [comment]
**Thank you for taking the time to report this issue.**

As part of a repository cleanup, we are closing issues that have been open for more than 12 months without resolution. A **beta version** of the plugin has been released that includes significant changes and improvements.

We encourage you to try the latest beta and see if your issue has been addressed. If the problem persists, please feel free to open a new issue with updated details and logs from the latest version.

Thank you for your patience and support!

**Comment by ArtB23** [comment]
I’m reporting that my issue seems to be resolved on S120 after a particular upgrade/downgrade sequence.

I first installed the latest beta 4.4.2beta43, then downgraded back to 4.3.8. After doing this, the cameras started working correctly again.

Current environment:

Plugin version: 4.3.8 (after downgrade)

iOS version: 26.3

Now the cameras open the live stream in real time when I click on them, just like other compatible cameras. Before this, the stream would not open correctly.

I have not yet tested snapshot refresh or other advanced features, but at least live view is now working reliably.

For reference, here is my config (with sensitive data removed):

{
    "platform": "EufySecurity",
    "name": "EufySecurity",
    "username": "xxxxxxxxxxxxx",
    "password": "xxxxxxxxxxxxx",
    "deviceName": "",
    "enableDetailedLogging": false,
    "omitLogFiles": false,
    "CameraMaxLivestreamDuration": 30,
    "pollingIntervalMinutes": 10,
    "hkHome": 1,
    "hkAway": 0,
    "hkNight": 3,
    "hkOff": 63,
    "ignoreStations": [
        "T81Axxxxxxxxxxxxxx",
        "T81A0xxxxxxxxxxxxx"
    ],
    "ignoreDevices": [],
    "country": "IT",
    "stations": [],
    "cameras": [
        {
            "serialNumber": "T81Axxxxxxxxxxx",
            "rtsp": false,
            "videoConfig": {
                "audio": false,
                "maxWidth": 1920,
                "maxHeight": 1080,
                "maxFPS": 15,
                "useSeparateProcesses": true
            },
            "enableButton": false,
            "motionButton": false,
            "lightButton": true,
            "refreshSnapshotIntervalMinutes": 30
        },
        {
            "serialNumber": "T81A0xxxxxxxxxxxxxx",
            "rtsp": false,
            "videoConfig": {
                "audio": false,
                "maxWidth": 1920,
                "maxHeight": 1080,
                "maxFPS": 15
            },
            "enableButton": false,
            "motionButton": false,
            "lightButton": true,
            "delayCameraSnapshot": false,
            "refreshSnapshotIntervalMinutes": 30
        }
    ],
    "cleanCache": true,
    "ignoreMultipleDevicesWarning": false,
    "autoSyncStation": false,
    "enableEmbeddedPKCS1Support": true,
    "_bridge": {
        "username": "xxxxxxxxx",
        "port": xxxxxxxxxxx
    }
}


Maybe the beta updated something in cache/firmware/auth that persisted after the downgrade.

Hope this helps others or the developers with troubleshooting 👍

**Comment by lenoxys** [comment]
@ArtB23 Thanks for your feedback, 4.4.2 will be released soon so you won't need to do back and forth and try to figure out.
